dts/arm64/nxp/nxp_mimx943_a55.dtsi

```diff
@@ -192,6 +192,85 @@
 		status = "disabled";
 	};
 
+	gpio2: gpio@43810000 {
+		compatible = "nxp,imx-rgpio";
+		reg = <0x43810000 DT_SIZE_K(64)>;
+		interrupts = <GIC_SPI 54 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>,
+				<GIC_SPI 55 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>;
+		interrupt-names = "irq_0", "irq_1";
+		interrupt-parent = <&gic>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		ngpios = <32>;
+		status = "disabled";
+	};
+
+	gpio3: gpio@43820000 {
+		compatible = "nxp,imx-rgpio";
+		reg = <0x43820000 DT_SIZE_K(64)>;
+		interrupts = <GIC_SPI 56 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>,
+				<GIC_SPI 57 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>;
+		interrupt-names = "irq_0", "irq_1";
+		interrupt-parent = <&gic>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		ngpios = <26>;
+		status = "disabled";
+	};
+
+	gpio4: gpio@43840000 {
+		compatible = "nxp,imx-rgpio";
+		reg = <0x43840000 DT_SIZE_K(64)>;
+		interrupts = <GIC_SPI 58 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>,
+				<GIC_SPI 59 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>;
+		interrupt-names = "irq_0", "irq_1";
+		interrupt-parent = <&gic>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		ngpios = <32>;
+		status = "disabled";
+	};
+
+	gpio5: gpio@43850000 {
+		compatible = "nxp,imx-rgpio";
+		reg = <0x43850000 DT_SIZE_K(64)>;
+		interrupts = <GIC_SPI 60 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>,
+				<GIC_SPI 61 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>;
+		interrupt-names = "irq_0", "irq_1";
+		interrupt-parent = <&gic>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		ngpios = <32>;
+		status = "disabled";
+	};
+
+	gpio6: gpio@43860000 {
+		compatible = "nxp,imx-rgpio";
+		reg = <0x43860000 DT_SIZE_K(64)>;
+		interrupts = <GIC_SPI 62 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>,
+				<GIC_SPI 63 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>;
+		interrupt-names = "irq_0", "irq_1";
+		interrupt-parent = <&gic>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		ngpios = <32>;
+		status = "disabled";
+	};
+
+	gpio7: gpio@43870000 {
+		compatible = "nxp,imx-rgpio";
+		reg = <0x43870000 DT_SIZE_K(64)>;
+		interrupts = <GIC_SPI 64 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>,
+				<GIC_SPI 65 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>;
+		interrupt-names = "irq_0", "irq_1";
+		interrupt-parent = <&gic>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		ngpios = <28>;
+		gpio-reserved-ranges = <10 6>;
```

Comment:
I see you use the `gpio-reserved-ranges` feature. However I don't see it implemented in GPIO ISR handler.
These reserved pins can still trigger an interrupt, I would filter these already out in the ISR otherwise an assertion can hit.

Reply:
Hi, @PetervdPerk-NXP,  could you please help me to understand why reverved pins can still trigger an interrupt? in RGPIO driver, "mcux_rgpio_pin_interrupt_configure" is used to configure interrupt for each pin, and in this function it has already use port_pin_mask filter reserved pins, that means reverved pins ICR(Interrupt Control Register)'s IRQC will be keeped as reset value "0000": ISF is disabled, so that it ISF bit for reserved pins will keep as 0 and "base->ISFR[0] = int_flags" in mcux_rgpio_port_isr() will not have the interrupt active flags for reserved pins. so in a word, we don't worry about reverved pins can trigger interrupt in current device driver.

Reply:
Let's say the M33 and A55 core both run a Zephyr kernel.

Buth both allocate the GPIO1.
M33 use does `gpio-reserved-ranges = <0 16>;`
A55 use does `gpio-reserved-ranges = <16 16>;`

If the A55 for example enables an IRQ on GPIO1 Pin 1 then it will write the ICR register setting Pin. When GPIO1 peripheral triggers an interrupt, then both A55 and M33 will receive the interrupt, but the M33 will forward the interrupt as well to zephyr kernel even though it's reserved.